src/main/scala/beam/agentsim/infrastructure/charging/ChargingInquiryData.scala

```diff
@@ -0,0 +1,16 @@
+package beam.agentsim.infrastructure.charging
+
+case class ChargingInquiryData(data: Map[ChargingPoint, ChargingPreference]) {
+  val agentMustCharge: Boolean = data.values.exists(_ == ChargingPreference.MustCharge)
+}
+
+object ChargingInquiryData {
```

Comment:
Interesting - I had missed [that you can provide a custom apply that overrides the built in one directly](https://github.com/scala/scala/pull/5816). I really thought this wouldn't compile - which I guess it wouldn't before that fix :p

Reply:
ya! i agree, and i don't know if maybe the scala compiler should instead warn about this, since i don't think this behavior is defined. 

this code isn't referenced anywhere yet, and is just a placeholder for things we have been working on.